### PR TITLE
[WIP] Proposal for a Gofer 4.x

### PR DESCRIPTION
**Pull request by jkrems**
- [x] Use modern syntax (node 8+).
- [x] Drop bluebird dependency.
- [x] Drop callback support.
- [x] Drop prototype magic and use class syntax.
- [ ] Use `URL` API over `url` module.
- [ ] Drop `qs` module, use `URLSearchParams`.

---
_This PR was started by: [git wf pr](https://github.groupondev.com/InteractionTier/workflow-cli/releases/tag/v1.7.0)_

### CODE REVIEW

**Comment by jkrems** [comment]
Suggestion by @dbushong: Go with 8+ immediately. This gives us the native `URL` support and we don't have to deal with a soon-to-be-non-LTS node 6.

**Comment by jkrems** [comment]
Foo!

dbeffa80-2bc5-11e9-8a7f-f0c9418fc56f

File: BREAKING_v4.md [hunk 1]
```diff
@@ -0,0 +1,101 @@
+# Gofer v3
+
+The biggest change in 3.x is that `gofer` is now a real class
```
Comment (dbushong) [inline BREAKING_v4.md]: v4? 4.x?

**Comment by dbushong** [comment]
Wish I'd remembered/reviewed this before #103

**Comment by jkrems** [comment]
But looks like #103 has the most important bits, closing this for now. :)

**Pull request closed without merging**